transport/conn.go

```diff
@@ -307,6 +320,14 @@ func OpenConn(addr string, localAddr *net.TCPAddr, cfg ConnConfig) (*Conn, error
 	return WrapConn(tcpConn)
 }
 
+func (c *Conn) resetWriterConn() {
+	c.w.conn.Reset(c.conn)
+}
+
+func (c *Conn) setWriterReseter() {
```

Comment:
Can't we have only `setWriterReseter` with `resetWriterConn` inlined? Or does the latter one have other uses?